doc/ztp/inband_ztp.md

```diff
@@ -0,0 +1,246 @@
+- Read oper state of all in-band interfaces (read PORT_TABLE in app DB) and if there is a link up change, perform restart to interfaces-config service, this will start DHCP discovery on all interfaces.
+- DHCP process starts.
+- DHCP hook /etc/dhcp/dhclient-enter-hooks.d/inband-ztp-ip sets the offered IP address on the in-band interface.
+- DHCP hook /etc/dhcp/dhclient-exit-hooks.d/ztp parses the option and writes it to a file in the filesystem.
+- Run discovery again, this time we expect that one of the options was parsed, we read the url and download the ZTP json. This is the the order of precedence:
+1. ZTP json file exist as part of the SW image.
+2. Download ZTP json specified in DHCP Option-67.
+3. Download ZTP json specified in DHCPv6 Option-59.
+4. Download simple provisioning script specified in DHCP Option-239.
+5. Download simple provisioning script specified in DHCPv6 Option-239.
+6. Download Minigraph xml and ACL json specified in DHCP Options 225 and 226.
+
+- Process the configuration sections appear in ZTP json (/host/ztp/ztp_data.json). Each section contains plugin to execute. There are several types of plugins:
+1. configdb-json - the plugin is used to download config DB json file and apply it. A config reload is performed during which various SWSS services may restart.
+2. firmware - this plugin is used for image management on a switch. It can be used to install, remove and boot selection of images. sonic_installer utility is used by this plugin to perform the supported functions.
+3. connectivity-check - this plugin is used to ping a remote host and verify if the switch is able to reach the remote host.
+4. snmp - this plugin is used to configure SNMP community string on.
+5. script - this plugin is used to run a user script.
+
+In the below example, there are 3 sections to process:
+```
+{
+	"ztp": {
+		"01-configdb-json": {
+			"url": {
+				"source": "http://192.168.0.1/ZTP/config_db.json",
+				"destination": "/etc/sonic/config_db.json"
+			}
+		},
+		"02-configdb-json": {
+			"url": {
+				"source": "http://192.168.0.1/ZTP/config_db-FEATURE.json",
+				"destination": "/etc/sonic/config_db.json"
+			},
+			"dynamic-url": {
+				"source": {
+					"prefix": "http://192.168.0.1/ZTP/",
+					"identifier": "hostname",
+					"suffix": "_config_db.json"
+				},
+				"destination": "/etc/sonic/config_db.json"
+			}
+		},
+		"03-script": {
+			"plugin": {
+				"url": {
+					"source": "http://192.168.0.1/ZTP/03-script.sh"
+				}
+			},
+			"reboot-on-failure": "true"
+		}
+	}
+}
+```
+- Last step is to clear ZTP configuration from the switch:
+1. Clear ZTP and COPP tables from config DB using sonic-db-cli.
+2. Clear ZTP and COPP tables from config_db.json.
+3. Delete file ztp_dhcp.json and restart interfaces-config service. When ztp_dhcp.json does not exist, interfaces-config service will create dhclient.conf without request of ZTP options.
```

Comment:
"interfaces-config service will create dhclient.conf without request of ZTP options."
What dose the above sentence mean?
My understand It will not send dhcp discovery. Is it correct?

Reply:
It will send DHCP discovery but without request for ZTP options (67, 239 etc..)